### DISCUSSION TITLE
Retrieving the list of media files from the database

### DISCUSSION THREAD

**Discussion by kharaei** [OrchardCMS/OrchardCore#16855] [Q&A] [answered]
Hello

I create a custom class for the IMediaCreatingEventHandler interface to upload media files to a minIO server and store their names in the database. (Due to security reasons, the minIO server does not return the list of all files.) However, I couldn't get the media section to read and display the list of files from the database.

Could you please guide me on how to implement this step?

**Comment by hishamco** [comment]
While this is a custom implementation, I prefer to have a look to the current media store implementations

**Comment by kharaei** [comment]
Thank you...
I created this class :
```
public class CustomFileStore(
    IFileStore fileStore,
    string requestBasePath,
    string cdnBaseUrl,
    IEnumerable<IMediaEventHandler> mediaEventHandlers,
    IEnumerable<IMediaCreatingEventHandler> mediaCreatingEventHandlers,
    ILogger<CustomFileStore> logger)
    : DefaultMediaFileStore(fileStore, requestBasePath, cdnBaseUrl, mediaEventHandlers, mediaCreatingEventHandlers,
        logger)
{
    public override IAsyncEnumerable<IFileStoreEntry> GetDirectoryContentAsync(string path = null, bool includeSubDirectories = false)
    {
        logger.LogInformation("------------------------");
        logger.LogInformation("now");
        logger.LogInformation("------------------------");
        return AsyncEnumerable.Empty<IFileStoreEntry>();
    }
}
```

and injected it into the prooject:
`builder.Services.AddScoped<IMediaFileStore, CustomFileStore>();
`

 but it has no effect on the file manager!
 If possible, please guide me to solve the issue.

**Reply by hishamco** [reply]
The above code will return an empty list

**Reply by kharaei** [reply]
> The above code will return an empty list

Yes, I know. That was just an example. 
I meant that it doesn't enter this method. in fact, it doesn't work.

**Reply by hishamco** [reply]
Create a custom module and then register the service, if you register it inside the host then one of media modules enabled the latest registered service will win

**Reply by kharaei** [reply]
> Create a custom module and then register the service, if you register it inside the host then one of media modules enabled the latest registered service will win

Thank you very much for your response. I thought there might be a way to override it,
but it seems there’s no choice other than creating a custom module.

**Reply by hishamco** [reply, marked as answer]
This is how the ASP.NET Core DI works the last registered service is win, so that's why I'm suggesting to create a new module then register your service in the `Startup` file
